file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs

```diff
@@ -151,85 +151,137 @@ protected internal override async Task OnExecuteCommand()
-    {
-                            fileUploadProgressViewModel.CompleteBlocks = progress.blocksComplete;
-                            fileUploadProgressViewModel.TotalBlocks = progress.totalBlockCount;
-                        });
-                            if (fileUploadProgressViewModel.CompleteBlocks == fileUploadProgressViewModel.TotalBlocks)
+                            cancellationToken.ThrowIfCancellationRequested();
+                            await Task.WhenAll(
+                            Files.SelectMany(f => f.Files.Select(file => (f, file)))
+                            .Select(f =>
+                            {
+                                logger.LogInformation("File Share Service upload files started for file:{file} and BatchId:{BatchId} .", f.file.Name, batchHandle?.BatchId);
+                                var fileUploadProgressViewModel = new FileUploadProgressViewModel(f.file.Name);
+                                FileUploadProgress.Add(fileUploadProgressViewModel);
+
+                                var (newBatchFilesViewModel, file) = f;
+                                var openRead = fileSystem.File.OpenRead(file.FullName);
+                                return fileShareClient.AddFileToBatch(batchHandle, openRead, file.Name,
+        newBatchFilesViewModel.MimeType,
+        progress =>
+        {
+            Application.Current.Dispatcher.Invoke(() =>
+{
+    fileUploadProgressViewModel.CompleteBlocks = progress.blocksComplete;
+    fileUploadProgressViewModel.TotalBlocks = progress.totalBlockCount;
+});
+            if (fileUploadProgressViewModel.CompleteBlocks == fileUploadProgressViewModel.TotalBlocks)
+            {
+                logger.LogInformation("File Share Service upload files completed for file:{file} and BatchId:{BatchId} .", f.file.Name, batchHandle.BatchId);
+            }
+        }, cancellationToken, newBatchFilesViewModel.Attributes?.Select(k => new KeyValuePair<string, string>(k.Key, k.Value)).ToArray())
+                                .ContinueWith(
+                                            res =>
+                                            {
+                                                if (res.Result.IsSuccess)
+                                                {
+                                                    ExecutionResult = $"File Share Service add file to batch completed for file:{f.file.Name} and BatchId:{batchHandle?.BatchId}";
+                                                    logger.LogInformation("File Share Service add file to batch completed for file:{file} and BatchId:{BatchId} .", f.file.Name, batchHandle?.BatchId);
+                                                }
+                                                else if (res.Result.Errors != null && res.Result.Errors.Any())
+                                                {
+                                                    ExecutionResult = (res.Result.Errors != null && res.Result.Errors.Any()) ?
```

Comment:
This code pattern for populating ExecutionResult from IResult is repeated in lots of places. Please extract to a new method:
```
private string GetErrors<T>(FileShareAdminClient.Models.Response.IResult<T> apiResult, string altMessage)
        {
            return (apiResult.Errors != null && apiResult.Errors.Any()) ?
                    string.Join(Environment.NewLine, apiResult.Errors.Select(e => e.Description)) :
                    altMessage;
        }
        
                        ExecutionResult = GetErrors(setBatchExpiryResult,
                    $"File Share Service set expiry date failed for batch ID:{ batchHandle?.BatchId} with status: {setBatchExpiryResult.StatusCode}.");
```